oneflow/core/functional/impl/math_functor.cpp

```diff
@@ -269,6 +269,78 @@ class TransposeFunctor {
   std::shared_ptr<OpExpr> op_;
 };
 
+class MovedimVecFunctor {
+ public:
+  MovedimVecFunctor() = default;
+  static Maybe<void> CheckNoRepeat(const std::vector<int32_t>& perm, std::vector<int32_t> &perm_out, int32_t indim, const std::string &desc) {  
+    std::vector<bool> is_used(indim, false);
+    FOR_RANGE(size_t, i, 0, perm.size()) {
+      int32_t item = perm[i];
+      if(item < 0){
+        item += indim;
+      }
+      CHECK_GE_OR_RETURN(item, 0) << ", Dimension out of range (expected to be in range of [" << -indim <<", " <<indim-1<< "], but got "<< perm[i] << ")";
+      CHECK_LT_OR_RETURN(item, indim) << ", Dimension out of range (expected to be in range of [" << -indim <<", " << indim-1 << "], but got " << perm[i] << ")";;
+      CHECK_EQ_OR_RETURN(is_used[item], false)<<"repeated dim in "<<desc;
```

Comment:
```suggestion
      CHECK_EQ_OR_RETURN(is_used[item], false) << "repeated dim in " << desc;
```

我比较好奇 `make of_format` 原来无法检查到这类“拥挤”的写法。 不过运算符之间是否留空格，其实各个编码规范里都会描述的，你可以找一本参考下。或者直接参考 oneflow 已有的代码。

Reply:
刚才简单测试了一下，make of_format是可以修正这种“拥挤”的写法的。我猜可能是因为没有跑ci，也就没有自动format